receiver/snmpreceiver/scraper.go

```diff
@@ -102,16 +104,12 @@ func (s *snmpScraper) scrapeScalarMetrics(
 	if len(scalarData) == 0 {
 		return
 	}
-
-	// Create general resource if we're about to make scalar metrics
-	resource := metricHelper.getResource(generalResourceKey)
-	if resource == nil {
-		metricHelper.createResource(generalResourceKey, map[string]string{})
-	}
+	// Retrieve scalar OID SNMP data for resource attributes
+	scalarOIDScalarResourceAttributeValues := s.scrapeScalarResourceAttributes(configHelper.getResourceAttributeScalarOIDs(), scraperErrors)
```

Comment:
nit: The variable name here seems longer than is helpful.
```suggestion
	scalarResourceAttributes := s.scrapeScalarResourceAttributes(configHelper.getResourceAttributeScalarOIDs(), scraperErrors)
```